source/util.cpp

```diff
@@ -99,20 +100,35 @@ string indent(string const &code, string const &indentation)
 	return r;
 }
 
+std::string join( std::vector<std::string>& splits, std::string const& sep ) {
+	if ( !splits.empty() ) {
+		std::ostringstream ss;
+		ss << splits[0];
+		for ( unsigned j=1; j<splits.size(); ++j ) {
+			ss << sep << splits[j];
+		}
+  	return ss.str();
+	}
+	return std::string();
```

Comment:
I think we can simplify this a bit if we use STL algorithms, how about something like:
```C++
std::ostringstream o;
std::copy(splits.begin(), splits.end(), std::ostream_iterator<std::string>(o, sep));
std::string r = o.str();
r.resize(r.size() - sep.size());
return r;
```